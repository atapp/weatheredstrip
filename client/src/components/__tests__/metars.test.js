import React from 'react';
import { render, cleanup } from 'react-testing-library'
import ReactDOM from 'react-dom';
import Metars from '../metars.react';
import 'jest-dom/extend-expect';
import testData from './test_data.json';
afterEach(cleanup)

testData.map(airport => {
  it('metar renders properly', () => {
    const { container } = render(<Metars data={ airport[ 'METAR' ] }/>)
    const metars = container.firstChild
    expect(metars).toMatchSnapshot()
  });
})

it('empty metar renders nothing', () => {
  const { container } = render(<Metars data={ [] }/>)
  const metars = container.firstChild
  expect(metars).toMatchSnapshot()
})

it('empty metar renders nothing', () => {
  const { container } = render(<Metars />)
  const metars = container.firstChild
  expect(metars).toMatchSnapshot()
})
