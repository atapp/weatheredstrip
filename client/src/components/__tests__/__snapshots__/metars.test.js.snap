// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`metar renders properly 1`] = `
<div
  class="Metars"
>
  <div
    class="notif-text"
  >
    METAR CYMX 172100Z 18005KT 170V230 35SM FEW250 M17/M24 A3033 RMK CI2 SLP280
  </div>
  <div
    class="notif-text"
  >
    METAR CYMX 172000Z 21005KT 180V260 35SM FEW250 M16/M24 A3034 RMK CI2 SLP286
  </div>
  <div
    class="notif-text"
  >
    METAR CYMX 171900Z 19007KT 45SM FEW220 FEW280 M17/M24 A3036 RMK CI1CI1 CI TR CI TR SLP292
  </div>
</div>
`;

exports[`metar renders properly 2`] = `
<div
  class="Metars"
>
  <div
    class="notif-text"
  >
    METAR CYUL 172100Z VRB02KT 30SM FEW160 SCT240 M15/M24 A3035 RMK AC1CI3 SLP282
  </div>
  <div
    class="notif-text"
  >
    METAR CYUL 172000Z 22006KT 30SM FEW240 M15/M23 A3037 RMK CI2 SLP290
  </div>
  <div
    class="notif-text"
  >
    METAR CYUL 171900Z 24007KT 30SM FEW240 M15/M23 A3039 RMK CI1 SLP297
  </div>
</div>
`;
